File: GuiaApe06/Ejercicio9.java
import java.util.LinkedList; import java.util.Scanner; 
  
class Turno {     private int numero; 
  
    public Turno(int numero) {         this.numero = numero; 
    } 
  
    public void setTurno(int numero) {         this.numero = numero; 
    } 
  
    public int getTurno() {         return numero; 
    } 
} 
  
public class Ejercicio9 { 
  
    public static void agregarTurno(LinkedList<Turno> turnos, int[] siguienteNumero) {         turnos.addLast(new Turno(siguienteNumero[0])); 
        System.out.println("Se agreog el turno #" + siguienteNumero[0]);         siguienteNumero[0]++; 
    } 
  
    public static void mostrarTurnos(LinkedList<Turno> turnos) {         if (turnos.isEmpty()) { 
            System.out.println("No hay turnos en espera.");             return; 
        } 
  
        System.out.println("\nTurnos en espera:");         for (Turno t : turnos) { 
            System.out.println("Turno #" + t.getTurno()); 
        } 
    } 
  
    public static void atenderTurno(LinkedList<Turno> turnos) {         if (turnos.isEmpty()) { 
            System.out.println("No hay turnos para atender.");             return; 
        } 
  
        System.out.println("Atendiendo turno #" + turnos.getFirst().getTurno());         turnos.removeFirst(); 
    } 
  
    public static void cancelarTurno(LinkedList<Turno> turnos, int numeroBuscado) { 
        for (int i = 0; i < turnos.size(); i++) {             if (turnos.get(i).getTurno() == numeroBuscado) {                 turnos.remove(i); 
                System.out.println("Turno #" + numeroBuscado + " cancelado.");                 return; 
            } 
        } 
  
        System.out.println("Turno no encontrado."); 
    } 
  
    public static void main(String[] args) { 
        LinkedList<Turno> turnos = new LinkedList<>(); 
        Scanner sc = new Scanner(System.in); 
  
        int[] siguienteNumero = {1};         int opcion;         int numeroCancelar; 
  
        do { 
            System.out.println("\n-SISTEMA DE TURNOS -"); 
            System.out.println("1. Agregar turno"); 
            System.out.println("2. Atender turno"); 
            System.out.println("3. Mostrar turnos"); 
            System.out.println("4. Cancelar turno"); 
            System.out.println("5. Salir"); 
            System.out.print("Seleccione una opción: ");             opcion = sc.nextInt(); 
  
            switch (opcion) {                 case 1: 
                    agregarTurno(turnos, siguienteNumero);                     break; 
  
                case 2: 
                    atenderTurno(turnos);                     break; 
  
                case 3: 
                    mostrarTurnos(turnos);                     break; 
  
                case 4: 
                    System.out.print("Ingrese el nmero del turno a cancelar: ");                     numeroCancelar = sc.nextInt();                     cancelarTurno(turnos, numeroCancelar);                     break; 
  
                case 5: 
 
                    System.out.println("Saliendo del sistema...");                     break; 
  
                default: 
                    System.out.println("Opcio invalida"); 
            } 
  
        } while (opcion != 5); 
  
        sc.close(); 
    } 
}   

File: GuiaApe06/Ejercicio9.cpp
#include <iostream>
#include <list>
using namespace std;

class Turno {
private:
    int numero;

public:
    Turno(int n = 0) {
        numero = n;
    }

    void setTurno(int n) {
        numero = n;
    }

    int getTurno() const {
        return numero;
    }
};

// Agregar un turno
void agregarTurno(list<Turno>& turnos, int& siguienteNumero) {
    Turno nuevo(siguienteNumero);
    turnos.push_back(nuevo);
    cout << "Se agrego el turno #" << siguienteNumero << endl;
    siguienteNumero++;
}

// Mostrar turnos
void mostrarTurnos(const list<Turno>& turnos) {
    if (turnos.empty()) {
        cout << "No hay turnos en espera.\n";
        return;
    }

    cout << "\nTurnos en espera:\n";
    for (const Turno& t : turnos) {
        cout << "Turno #" << t.getTurno() << endl;
    }
}

// Atiende al primer turno
void atenderTurno(list<Turno>& turnos) {
    if (turnos.empty()) {
        cout << "No hay turnos para atender.\n";
        return;
    }

    cout << "Atendiendo turno #" << turnos.front().getTurno() << endl;
    turnos.pop_front();
}

// Quitar un turno
void cancelarTurno(list<Turno>& turnos, int numeroBuscado) {
    for (auto it = turnos.begin(); it != turnos.end(); ++it) {
        if (it->getTurno() == numeroBuscado) {
            turnos.erase(it);
            cout << "Se cancelo el turno #" << numeroBuscado << endl;
            return;
        }
    }

    cout << "El turno #" << numeroBuscado << " no existe.\n";
}

int main() {
    list<Turno> turnos;
    int siguienteNumero = 1;
    int opcion, numeroCancelar;

    do {
        cout << "\n===== SISTEMA DE TURNOS =====\n";
        cout << "1. Agregar turno\n";
        cout << "2. Atender turno\n";
        cout << "3. Mostrar turnos\n";
        cout << "4. Cancelar turno\n";
        cout << "5. Salir\n";
        cout << "Seleccione una opcion: ";
        cin >> opcion;

        switch (opcion) {
            case 1:
                agregarTurno(turnos, siguienteNumero);
                break;
            case 2:
                atenderTurno(turnos);
                break;
            case 3:
                mostrarTurnos(turnos);
                break;
            case 4:
                cout << "Ingrese el numero del turno a cancelar: ";
                cin >> numeroCancelar;
                cancelarTurno(turnos, numeroCancelar);
                break;
            case 5:
                cout << "Saliendo del sistema...\n";
                break;
            default:
                cout << "Opcion invalida.\n";
        }

    } while (opcion != 5);

    return 0;
} 
